### PR TITLE
pull my sprite code outta dumpsta

### PR DESCRIPTION
**Pull request by RemoveRusky**

### CODE REVIEW

File: ENIGMAsystem/SHELL/Universal_System/spritestruct.cpp [hunk 1]
```diff
@@ -17,687 +17,460 @@
...
 
-    spr->texturearray.clear();
-    spr->texturexarray.clear();
-    spr->textureyarray.clear();
-    spr->texturewarray.clear();
-    spr->textureharray.clear();
-  enigma::sprite_add_to_index(spr, filename, imgnumb, precise, transparent,
-      smooth, x_offset, y_offset, mipmap);
-    return true;
+  delete pixels;
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/spritestruct.cpp]: `delete[]`

File: ENIGMAsystem/SHELL/Universal_System/spritestruct.cpp [hunk 2]
```diff
@@ -17,687 +17,460 @@
...
+
+  b[1].bottom = b[0].bottom - yoffset;
+  b[1].left = b[0].left - xoffset;
+  b[1].top = b[0].top - yoffset;
+  b[1].right = b[0].right - xoffset;
+
+  return b;
+}
+
+inline void addTexture(Sprite *spr, unsigned char *pxdata, unsigned fullWidth, unsigned fullHeight,
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/spritestruct.cpp]: You're computing `fullWidth`/`fullHeight` now; why are you also passing them as read-only?

File: ENIGMAsystem/SHELL/Universal_System/spritestruct.cpp [hunk 3]
```diff
@@ -17,687 +17,460 @@
...
+      }
+    }
 
-bool sprite_replace(int ind, string filename, int imgnumb, bool precise, bool transparent, bool smooth, bool preload, int x_offset, int y_offset, bool free_texture, bool mipmap)
-{
-    enigma::sprite *spr;
-    if (!get_sprite_mtx(spr, ind))
-        return false;
+    SubImage s;
+    s.textureID = graphics_create_texture(targetWidth, targetHeight, fullcellwidth, fullHeight, pixels, mipmap);
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/spritestruct.cpp]: You're passing `fullcellwidth`, to which you've arbitrarily added `1`.

File: ENIGMAsystem/SHELL/Universal_System/spritestruct.cpp (hunk 1 above)
Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Universal_System/spritestruct.cpp]: Done: b3610d24617b952ed68f6b97a5dc0cc0e0ef7d52

**Comment by RobertBColton** [comment]
Alright, I fixed the other issue now with the texture ids: 918337d84d1d7a0ad4aaa9ae3e77576f3c37ce

You need to have sprites init allocate the *ENTIRE* sprites vector at once to avoid multiple reallocations from push_back, this can shave seconds off of a game's load time. Second, the issue was basically that you didn't have the sprite ids matching what was written to the exe, so the code was referring to the wrong sprites, hence all the wrong textures before.

Anyway, that's it, also notice the shotgun sprite is still smushed, something with the widths.
FPS Game Sprites Cleanup

**Comment by RobertBColton** [comment]
Alright, I fixed the other issue now with the texture ids: 15b8069c2b878021a54a2d05861647d15e8fd4ad

You need to have sprites init allocate the *ENTIRE* sprites vector at once to avoid multiple reallocations from push_back, this can shave seconds off of a game's load time. Second, the issue was basically that you didn't have the sprite ids matching what was written to the exe, so the code was referring to the wrong sprites, hence all the wrong textures before.

Anyway, that's it, also notice the shotgun sprite is still smushed, something with the widths.
FPS Game Sprites Cleanup

**Comment by gfundies** [comment]
This has gone completely stale its easier to just redo it

**Comment by RobertBColton** [comment]
I agree with you here. I was going to fix the conflicts by recreating the branch and cherry-picking the commits but ran into conflicts. I decided not to fix the conflicts at this time because they are in functions like `sprite_create_from_screen` and such which I started generalizing in #1443. If we generalize those functions and clean them up first, then this pull request actually becomes a lot easier to do.

**Pull request closed without merging**